File: testdata/fixtures/java-clean/src/main/java/com/example/handler/OrderHandler.java
// Handler layer: the edge. May import service, stdlib and external deps.
package com.example.handler;

import java.io.IOException;

import com.google.gson.Gson;

import com.example.service.OrderService;

public final class OrderHandler {
    private final OrderService service = new OrderService();
    private final Gson gson = new Gson();

    public String handle(int total) throws IOException {
        return gson.toJson(service.placeOrder(total).id);
    }
}

File: testdata/fixtures/kotlin-clean/src/main/kotlin/com/example/handler/OrderHandler.kt
// Handler layer: the edge. May import service, stdlib and external deps.
package com.example.handler

import java.io.IOException

import com.squareup.moshi.Moshi

import com.example.service.OrderService

class OrderHandler {
    private val service = OrderService()
    private val moshi = Moshi.Builder().build()

    @Throws(IOException::class)
    fun handle(total: Int): String {
        val order = service.placeOrder(total)
        return moshi.adapter(Long::class.java).toJson(order.id)
    }
}
